### PR TITLE
Fix: create event from `Notify`

### PR DESCRIPTION
**Pull request by Jim8y**
Alternative to https://github.com/neo-project/neo-devpack-dotnet/pull/845

This pr will create `event` to manifest if user calls `Runtime.Notify`

### CODE REVIEW

**Comment by Jim8y** [comment]
@shargon @roman-khimov @cschuchardt88 @vncoelho @superboyiii  which one you think is better? Ban Notify or create event for notify?

**Comment by cschuchardt88** [comment]
Create Event, But do think the name should be changed to `Runtime.CreateEvent`

File: src/Neo.Compiler.CSharp/CompilationContext.cs [hunk 1]
```diff
@@ -66,6 +66,7 @@ public class CompilationContext
         internal int StaticFieldCount => staticFields.Count + anonymousStaticFields.Count + vtables.Count;
         private byte[] Script => script ??= GetInstructions().Select(p => p.ToArray()).SelectMany(p => p).ToArray();
 
+        public void AddEvent(AbiEvent abiEvent) => eventsExported.Add(abiEvent);
         static CompilationContext()
```
Comment (shargon) [inline src/Neo.Compiler.CSharp/CompilationContext.cs:70]:
```suggestion
        public void AddEvent(AbiEvent abiEvent) => eventsExported.Add(abiEvent);

        static CompilationContext()
```

File: src/Neo.Compiler.CSharp/MethodConvert.cs [hunk 2]
```diff
@@ -3985,7 +3985,14 @@ private void Call(SemanticModel model, IMethodSymbol symbol, ExpressionSyntax? i
             if (convert is null)
                 CallVirtual(symbol);
             else
+            {
                 EmitCall(convert);
+                // process events from calling Runtime.Notify
+                if (symbol.ToString().Contains("Services.Runtime.Notify"))
```
Comment (shargon) [inline src/Neo.Compiler.CSharp/MethodConvert.cs:3991]: StartWith or Equals?

File: src/Neo.Compiler.CSharp/AbiEvent.cs [hunk 3]
```diff
@@ -32,5 +34,27 @@ public AbiEvent(IEventSymbol symbol)
...
+                {
+                    Name = symbol.Parameters[1].Name,
+                    Type = ContractParameterType.ByteArray
+                }
+            };
+
+            return new AbiEvent(symbol, value, abiParameters);
+        }
+
     }
```
Comment (shargon) [inline src/Neo.Compiler.CSharp/AbiEvent.cs:59]:
```suggestion
    }
```

File: src/Neo.Compiler.CSharp/AbiEvent.cs [hunk 4]
```diff
@@ -1,20 +1,22 @@
...
 using Microsoft.CodeAnalysis;
 using Neo.SmartContract.Manifest;
 using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Neo.SmartContract;
 
 namespace Neo.Compiler
 {
-    class AbiEvent
+    public class AbiEvent
```
Comment (shargon) [inline src/Neo.Compiler.CSharp/AbiEvent.cs:19]: internal?

**Comment by Jim8y** [comment]
As is confirmed with Erik, will ban `Notify` directly. Close this one.

**Pull request closed without merging**